metrics/sanitize.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import (
+	"strings"
+	"unicode"
+)
+
+func SanitizeLabel(value string) string {
+	var builder strings.Builder
+	previousUnderscore := false
+
+	for _, char := range value {
+		if char == ' ' || char == '-' {
+			if !previousUnderscore {
+				builder.WriteRune('_')
+				previousUnderscore = true
+			}
+		} else if unicode.IsLetter(char) || unicode.IsDigit(char) || char == '_' {
```

Comment:
I agree with idea of sanitazing the labels - but why the logic is so complex?

can it be just
* if this is letter or digit - leave it
* Else - put `_`

We can also optimize and do not take `builder.String()` if there were no changes in `value` (input).

WDYT?

Reply:
what you propose - was a first straightforward implementation though

the implementation tries to conform [prometheus' data model](https://prometheus.io/docs/concepts/data_model/) rules:
```
Labels may contain ASCII letters, numbers, as well as underscores. They must match the regex [a-zA-Z_][a-zA-Z0-9_]*.
Label names beginning with __ (two "_") are reserved for internal use.
Label values may contain any Unicode characters.
```